compute/ibm_vpc.py

```diff
@@ -638,6 +598,225 @@ def _wait_until_vm_state(
 
         raise NodeError(f"{node_details['name']} is in {node_details['status']} state.")
 
+    @retry(DNS_RETRY_EXCEPTIONS, tries=3, delay=5)
+    def _list_resource_records_page(
+        self, dns_svc_id: str, dns_zone_id: str, limit: int, offset: int
+    ) -> Dict:
+        """
+        List a single page of resource records with retry support.
+
+        Args:
+            dns_svc_id (str):   GUID of the DNS Service.
+            dns_zone_id (str):  DNS Zone ID.
+            limit (int):        Number of records per page.
+            offset (int):       Offset for pagination.
+
+        Returns:
+            Dict: Response result containing resource records.
+
+        Raises:
+            ApiException: If the API call fails after retries.
+            ConnectionError: If connection fails after retries.
+            ReadTimeout: If request times out after retries.
+            RequestException: If request fails after retries.
+        """
+        resp = self.dns_service.list_resource_records(
+            instance_id=dns_svc_id,
+            dnszone_id=dns_zone_id,
+            limit=limit,
+            offset=offset,
+        )
+        return resp.get_result()
+
+    def _list_resource_records_with_pagination(
+        self, dns_svc_id: str, dns_zone_id: str, limit: int = 50
+    ) -> List[Dict]:
```

Comment:
Wdyt about using `iterator` instead this way we could avoid some of loops that we are introducing.